torchvision/datasets/kinetics.py

```diff
@@ -74,6 +148,89 @@ def __init__(self, root, frames_per_clip, step_between_clips=1, frame_rate=None,
         )
         self.transform = transform
 
+    def download_and_process_videos(self) -> None:
+        """Downloads all the videos to the _root_ folder in the expected format."""
+        tic = time.time()
+        self._download_videos()
+        toc = time.time()
+        print("Elapsed time for downloading in mins ", (toc - tic) / 60)
+        self._make_ds_structure()
+        toc2 = time.time()
+        print("Elapsed time for processing in mins ", (toc2 - toc) / 60)
+        print("Elapsed time overall in mins ", (toc2 - tic) / 60)
+
+    def _download_videos(self) -> None:
+        """download tarballs containing the video to "tars" folder and extract them into the _split_ folder where
+        split is one of the official dataset splits.
+
+        Raises:
+            RuntimeError: if download folder exists, break to prevent downloading entire dataset again.
+        """
+        if path.exists(self.split_folder):
+            raise RuntimeError(
+                f"The directory {self.split_folder} already exists. If you want to re-download or re-extract the images, "
+                f"delete the directory."
+            )
+        # check that the assignment was made properly
+        tar_path = path.join(self.root, "tars")
+        annotation_path = path.join(self.root, "annotations")
+        file_list_path = path.join(self.root, "files")
+
+        split_url = self._TAR_URLS[self.num_classes].format(split=self.split)
+        split_url_filepath = path.join(file_list_path, path.basename(split_url))
+        if not check_integrity(split_url_filepath):
+            download_url(split_url, file_list_path)
+        list_video_urls = open(split_url_filepath, "r")
+
+        if not check_integrity(path.join(annotation_path, f"{self.split}.csv")):
+            download_url(self._ANNOTATION_URLS[self.num_classes].format(split=self.split), annotation_path)
+        self.annotations = path.join(annotation_path, f"{self.split}.csv")
```

Comment:
Instead of setting this to `self` here, you could return it and pass it on to `_make_ds_structure`. Otherwise we still have the same problem as before: the attribute `annotations` only exists when `download=True` was used.